scripts/traverse.js

```diff
@@ -2,89 +2,145 @@
 const bcd = require('..');
 
 const { argv } = require('yargs').command(
-  '$0 <browser> [folder] [value]',
-  'Test for specified values in any specified browser',
+  '$0 [folder...]',
+  'Print feature names in the folder (and optionally filter features to specific browser or version values)',
   yargs => {
     yargs
-      .positional('browser', {
-        describe: 'The browser to test for',
-        type: 'string',
-      })
       .positional('folder', {
-        describe: 'The folder(s) to test (set to "all" for all folders)',
+        describe: 'The folder(s) to traverse',
+        type: 'array',
+        default: Object.keys(bcd).filter(k => k !== 'browsers'),
+      })
+      .option('browser', {
+        alias: 'b',
+        describe: 'Filter by a browser. May repeat.',
         type: 'array',
-        default: 'all',
+        nargs: 1,
+        default: Object.keys(bcd.browsers),
       })
-      .positional('value', {
-        describe: 'The value(s) to test against',
+      .option('filter', {
+        alias: 'f',
+        describe: 'Filter by version value. May repeat.',
         type: 'array',
-        default: ['null', 'true'],
+        nargs: 1,
+        default: [],
+      })
+      .option('non-real', {
+        describe:
+          'Filter to features with non-real values. Alias for "-f true -f null"',
+        type: 'boolean',
+        nargs: 0,
       })
       .option('depth', {
         alias: 'd',
         describe:
           'Depth of features to traverse (ex. "2" will capture "api.CSSStyleSheet.insertRule" but not "api.CSSStyleSheet.insertRule.optional_index")',
         type: 'number',
+        nargs: 1,
         default: 100,
-      });
+      })
+      .example(
+        'npm run traverse -- --browser=safari --non-real',
+        'Find all features containing non-real Safari entries',
+      )
+      .example(
+        'npm run traverse -- -b webview_android -f true',
+        'Find all features marked as true for WebVeiw',
+      )
+      .example(
+        'npm run traverse -- -b firefox -f 10',
+        'Find all features marked as supported since Firefox 10',
+      );
   },
 );
 
-function traverseFeatures(obj, depth, identifier) {
+function traverseFeatures(obj, browsers, values, depth, identifier) {
```

Comment:
How about a `iterateFeatures` using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function* that yields one entry a a time, and this `traverseFeatures` just uses `Array.from()` to return an array? The code that I have in mind will usually just want to iterate entries, having them all in an array would just be extra lines of code.

Reply:
I like this `iterateFeatures` idea too! 👍

Reply:
I'm not all that familiar with generator functions (this would be my first time writing one), and I'm not too certain on how to implement it into this code.  Having a function like that sounds like a good idea, though!  I'll see what I can whip up (and I'll talk about it a bit during my next meeting with foolip).

Reply:
Oops, we didn't end up talking about this when we met. It's basically a structure like this:
```js
function* iterateFeatures () {
  for (let i = 0; i < 10; i++) {
    yield i;
  }
}
```

Reply:
Hey @vinyldarkscratch, do you have questions about this? Or is this just waiting on you having time to complete it? If it's the former, let me know how I can help on this—I can write up a fuller generator example, if you'd like. If it's the latter, there's no rush here. Thank you!

Reply:
Slip of the mind on this one, actually -- I've been focusing my efforts on getting the real values up (along with the mdn-bcd-collector project, getting its coverage up more).  Spent a little bit of time on this again, and I think I got it!  I'm pushing a commit that should get the desired result, let me know how it looks?